src/sidebar/services/stream-filter.ts

```diff
@@ -1,31 +1,43 @@
 /**
  * Filter clause against which annotation updates are tested before being
  * sent to the client.
- *
- * @typedef FilterClause
- * @prop {'/group'|'/id'|'/references'|'/uri'} field
- * @prop {'equals'|'one_of'} operator
- * @prop {string|string[]} value
- * @prop {boolean} case_sensitive - TODO: Backend doesn't use this at present,
- *   but it seems important for certain fields (eg. ID).
  */
+type FilterClause = {
+  field: '/group' | '/id' | '/references' | '/uri';
+  operator: 'equals' | 'one_of';
+  value: string | string[];
 
-/**
- * @typedef Filter
- * @prop {string} match_policy - TODO: Remove this, the backend doesn't use it any more.
- * @prop {FilterClause[]} clauses
- * @prop {object} actions - TODO: Remove this, the backend doesn't use it any more.
- *  @prop {boolean} [actions.create]
- *  @prop {boolean} [actions.update]
- *  @prop {boolean} [actions.delete]
- */
+  /**
+   * @todo Backend doesn't use this at present, but it seems important for
+   *       certain fields (eg. ID).
+   */
+  case_sensitive: boolean;
+};
+
+type Filter = {
+  clauses: FilterClause[];
+
+  /**
+   * @deprecated
+   * @todo Remove this, the backend doesn't use it any more.
+   */
+  match_policy: string;
+
+  /**
+   * @deprecated
```

Comment:
I took the liberty to mark these as deprecated, since they had a comment saying they should be removed.